2018 Robot/src/org/usfirst/frc/team4999/utils/PIDFactory.java

```diff
@@ -0,0 +1,393 @@
+package org.usfirst.frc.team4999.utils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+import org.usfirst.frc.team4999.robot.RobotMap;
+import org.usfirst.frc.team4999.robot.sensors.GyroFusion;
+import org.usfirst.frc.team4999.robot.sensors.GyroFusion.Sensor;
+
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class PIDFactory {
+	
+	private static final String BASE = "/home/lvuser/";
+	private static final String MOVE_FILE = "movePID.properties";
+	private static final String TURN_FILE = "turnPID.properties";
+	private static final String TILT_FILE = "tiltPID.properties";
+	private static final String FAST_LIFT_FILE = "fastLiftPID.properties";
+	private static final String SLOW_LIFT_FILE = "slowLiftPID.properties";
+	private static final String ELBOW_FILE = "elbowPID.properties";
+	
+	private static final String DEFAULT_P = "0";
+	private static final String DEFAULT_I = "0";
+	private static final String DEFAULT_D = "0";
+	private static final String DEFAULT_ERR_ZONE = "10";
+	private static final String DEFAULT_TARGET_ZONE = "3";
+	private static final String DEFAULT_TARGET_TIME = "0.5";
+	
+	private static final PIDThread calculator = new PIDThread(7);
+	
+	public static void addToCalculator(MomentumPID cont) {
+		calculator.addController(cont);
+		if(!calculator.isAlive() && !calculator.isDead())
+			calculator.start();
+	}
+	
+	private static Properties openFile(String file) {
+		System.out.println("Loading " + file);
+		Properties ret = new Properties();
+		FileInputStream input = null;
+		try {
+			input = new FileInputStream(file);
```

Comment:
Consider formatting as a try-with-resources block.
try (FileInputStream input = new FileInputStream(file)) {
}
Then you don't need a finally clause.

https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html